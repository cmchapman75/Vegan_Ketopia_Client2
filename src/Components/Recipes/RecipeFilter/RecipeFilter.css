.cuisine_typeFilter, .recipe_typeFilter {
    height: 25px;
    font-size: 20px;
    padding-top: 5px;
    width: 280px;
    margin-top: 1.5%;
}

.cuisine_typeFilter {
    padding-bottom: -5px;;
}