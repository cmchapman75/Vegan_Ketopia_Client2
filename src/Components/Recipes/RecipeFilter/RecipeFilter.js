import React, { Component } from 'react';
import './RecipeFilter.css';

// child of RecipeSearch --creates dropdowns for Recipe Search Page
class RecipeFilter extends Component {
  render() {
    const { handleFilterChange, handleFilterChangeCuisine } = this.props;
    return (
      <div className="FilterOptions">
        <div className="FilterOptions__option">
          <label htmlFor="filter_recipeType">
            Recipe Type:
            <br></br>
            <select className="dropDown"
              // checked={filterOptions === "All"}
              onChange={e => handleFilterChange(e.target.value)}>
              <option value="all"></option>
              <option value="breakfast">Breakfast</option>
              <option value="lunch">Lunch</option>
              <option value="dinner">Dinner</option>
              <option value="dessert">Dessert</option>
              <option value="snack">Snack</option>
              <option value="Breads and Crackers">Breads and Crackers</option>
              <option value="Side Dish">Side Dish</option>
              <option value="fat-bombs">Fat Bombs</option>
            </select>
          </label>
        </div>
        <div className="FilterOptions__option">
          <label htmlFor="filter_cuisine_type">
            Cuisine Type:
            <br></br>
            <select className="dropDown"
                // checked={filterOptionsCuisine === "All"}
                onChange={e => handleFilterChangeCuisine(e.target.value)}>
                <option value="all"></option>
                <option value="Universal">Universal</option>
                <option value="American">American</option>
                <option value="Chinese">Chinese</option>
                <option value="French">French</option>
                <option value="German">German</option>
                <option value="Greek">Greek</option>
                <option value="Indian">Indian</option>
                <option value="Irish">Irish</option>
                <option value="Italian">Italian</option>
                <option value="Japanese">Japanese</option>                        
                <option value="Korean">Korean</option>
                <option value="Mexican">Mexican</option>
                <option value="Middle Eastern">Middle Eastern</option>
                <option value="Russian">Russian</option>                        
                <option value="Spanish">Spanish</option>               
                <option value="Thai">Thai</option>
                <option value="Vietnamese">Vietnamese</option>   
            </select>
          </label>
        </div>
      </div>
    );
  }
}

export default RecipeFilter;