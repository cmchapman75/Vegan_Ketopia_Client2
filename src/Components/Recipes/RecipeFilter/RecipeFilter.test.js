import React from 'react';
import ReactDOM from 'react-dom';
import RecipeFilter from './RecipeFilter';
import { BrowserRouter } from 'react-router-dom';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter>
    <RecipeFilter />
  </BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<BrowserRouter>
    <RecipeFilter />
  </BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});