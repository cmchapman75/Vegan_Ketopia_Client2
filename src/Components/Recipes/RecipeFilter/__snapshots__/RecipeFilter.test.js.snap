// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="FilterOptions"
>
  <div
    className="FilterOptions__option"
  >
    <label
      htmlFor="filter_recipeType"
    >
      Recipe Type:
      <select
        className="recipeTypeFilter"
        onChange={[Function]}
      >
        <option
          value={true}
        />
        <option
          value="breakfast"
        >
          Breakfast
        </option>
        <option
          value="lunch"
        >
          Lunch
        </option>
        <option
          value="dinner"
        >
          Dinner
        </option>
        <option
          value="dessert"
        >
          Dessert
        </option>
        <option
          value="snack"
        >
          Snack
        </option>
        <option
          value="Breads and Crackers"
        >
          Breads and Crackers
        </option>
        <option
          value="Side Dish"
        >
          Side Dish
        </option>
        <option
          value="fat-bombs"
        >
          Fat Bombs
        </option>
      </select>
    </label>
  </div>
  <div
    className="FilterOptions__option"
  >
    <label
      htmlFor="filter_cuisine_type"
    >
      Cuisine Type:
      <select
        className="cuisine_typeFilter"
        onChange={[Function]}
      >
        <option
          value={true}
        />
        <option
          value="Universal"
        >
          Universal
        </option>
        <option
          value="American"
        >
          American
        </option>
        <option
          value="Chinese"
        >
          Chinese
        </option>
        <option
          value="French"
        >
          French
        </option>
        <option
          value="German"
        >
          German
        </option>
        <option
          value="Greek"
        >
          Greek
        </option>
        <option
          value="Indian"
        >
          Indian
        </option>
        <option
          value="Irish"
        >
          Irish
        </option>
        <option
          value="Italian"
        >
          Italian
        </option>
        <option
          value="Japanese"
        >
          Japanese
        </option>
        <option
          value="Korean"
        >
          Korean
        </option>
        <option
          value="Mexican"
        >
          Mexican
        </option>
        <option
          value="Middle Eastern"
        >
          Middle Eastern
        </option>
        <option
          value="Russian"
        >
          Russian
        </option>
        <option
          value="Spanish"
        >
          Spanish
        </option>
        <option
          value="Thai"
        >
          Thai
        </option>
        <option
          value="Vietnamese"
        >
          Vietnamese
        </option>
      </select>
    </label>
  </div>
</div>
`;
